File: ports/csharp/src/core/tools/BcsGeometry.cs
namespace BCS.Core.Tools {
    // BcsPoint defines an X/Y coordinate
    public struct BcsPoint {
        public int X { get; set; }
        public int Y { get; set; }

        public BcsPoint(int x, int y) {
            X = x;
            Y = y;
        }
    }

    // BcsSize defines width and height dimensions
    public struct BcsSize {
        public int Width { get; set; }
        public int Height { get; set; }

        public BcsSize(int width, int height) {
            Width = width;
            Height = height;
        }
    }

    // BcsRect defines a 2D bounding box
    public struct BcsRect {
        public int X { get; set; }
        public int Y { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }

        public BcsRect(int x, int y, int width, int height) {
            X = x;
            Y = y;
            Width = width;
            Height = height;
        }

        public bool Contains(BcsPoint p) {
            return p.X >= X && p.X <= X + Width &&
                   p.Y >= Y && p.Y <= Y + Height;
        }

        public bool Intersects(BcsRect other) {
            return X < other.X + other.Width && X + Width > other.X &&
                   Y < other.Y + other.Height && Y + Height > other.Y;
        }
    }
}

File: ports/java/src/core/tools/BcsGeometry.java
package bcs.core.tools;

public class BcsGeometry {

    // BcsPoint defines an X/Y coordinate
    public static class BcsPoint {
        public int x;
        public int y;

        public BcsPoint(int x, int y) {
            this.x = x;
            this.y = y;
        }
    }

    // BcsSize defines width and height dimensions
    public static class BcsSize {
        public int width;
        public int height;

        public BcsSize(int width, int height) {
            this.width = width;
            this.height = height;
        }
    }

    // BcsRect defines a 2D bounding box
    public static class BcsRect {
        public int x;
        public int y;
        public int width;
        public int height;

        public BcsRect(int x, int y, int width, int height) {
            this.x = x;
            this.y = y;
            this.width = width;
            this.height = height;
        }

        public boolean contains(BcsPoint p) {
            return p.x >= this.x && p.x <= this.x + this.width &&
                   p.y >= this.y && p.y <= this.y + this.height;
        }

        public boolean intersects(BcsRect other) {
            return this.x < other.x + other.width && this.x + this.width > other.x &&
                   this.y < other.y + other.height && this.y + this.height > other.y;
        }
    }
}
